packages/headless/src/features/case-assist/case-assist-analytics-actions.ts

```diff
@@ -0,0 +1,243 @@
+import {createAsyncThunk} from '@reduxjs/toolkit';
+import {ThunkExtraArguments} from '../../app/store';
+import {handleOneAnalyticsEvent} from 'coveo.analytics';
+import {CaseAssistAppState} from '../../state/case-assist-app-state';
+import {
+  getClassificationById,
+  getDocumentSuggestionById,
+} from './case-assist-selectors';
+import {validatePayload} from '../../utils/validate-payload';
+import {NumberValue, StringValue} from '@coveo/bueno';
+
+export interface AsyncThunkOptions {
+  state: CaseAssistAppState;
+  extra: ThunkExtraArguments;
+}
+
+export const initializeTicketLogging = createAsyncThunk<
```

Comment:
I attempted to use the existing strongly typed `coveo.analytics` clients in headless and wasn't happy with the result. As you suggest, we'd need to expose some kind of Service client in `coveo.analytics` that makes it very easy to record Service events from headless or anywhere else.

> The reason it's done like that is because it adds all the boilerplate to be able to hook all the needed code to allow end users to "tap" into the request (preprocessRequest), so that they can modify/inspect the payload, and intercept the requests to log externally if need be.

That is a very good point indeed, and I wasn't aware it was a general Usage Analytics requirement.

> That was otherwise a problem with JSUI, where the "source of truth" about what is the correct metadata/events identifier was scattered everywhere around different projects, making it very hard to log client side analytics correctly if you were trying to do anything remotely custom.

One of the goals of the case assist controller is to make it easy for an implementer to log analytics events by keeping inputs to a minimum. But you are also right, that it would make sense to have a simplified service client provided directly by `coveo.analytics`.

Reply:
@olamothe The `coveo.analytics` concerns will be addressed in these stories
https://coveord.atlassian.net/browse/SFINT-3835
https://coveord.atlassian.net/browse/SFINT-3837

Reply:
Meanwhile, everything related to Usage Analytics has been removed.